homeassistant/components/forked_daapd/config_flow.py

```diff
@@ -0,0 +1,196 @@
+                            CONF_TTS_VOLUME, DEFAULT_TTS_VOLUME
+                        ),
+                    ): float,
+                    vol.Optional(
+                        CONF_LIBRESPOT_JAVA_PORT,
+                        default=self.config_entry.options.get(
+                            CONF_LIBRESPOT_JAVA_PORT, 24879
+                        ),
+                    ): int,
+                    vol.Optional(
+                        CONF_MAX_PLAYLISTS,
+                        default=self.config_entry.options.get(CONF_MAX_PLAYLISTS, 10),
+                    ): int,
+                }
+            ),
+        )
+
+
+def fill_in_schema_dict(some_input):
+    """Fill in schema dict defaults from user_input."""
+    schema_dict = {}
+    for field, _type in DATA_SCHEMA_DICT.items():
+        if some_input.get(str(field)):
+            schema_dict[
+                vol.Optional(str(field), default=some_input[str(field)])
+            ] = _type
+        else:
+            schema_dict[field] = _type
+    return schema_dict
+
+
+class ForkedDaapdFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a forked-daapd config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self):
+        """Initialize."""
+        self.discovery_schema = None
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Return options flow handler."""
+        return ForkedDaapdOptionsFlowHandler(config_entry)
+
+    async def validate_input(self, user_input):
+        """Validate the user input."""
+        websession = async_get_clientsession(self.hass)
+        validate_result = await ForkedDaapdAPI.test_connection(
+            websession=websession,
+            host=user_input[CONF_HOST],
+            port=user_input[CONF_PORT],
+            password=user_input[CONF_PASSWORD],
+        )
+        validate_result[0] = TEST_CONNECTION_ERROR_DICT.get(
+            validate_result[0], "unknown_error"
+        )
+        return validate_result
+
+    async def async_step_user(self, user_input=None):
+        """Handle a forked-daapd config flow start.
+
+        Manage device specific parameters.
+        """
+        if user_input is not None:
+            validate_result = await self.validate_input(user_input)
+            if validate_result[0] == "ok":  # success
+                _LOGGER.debug("Connected successfully. Creating entry")
+                # before creating entry, remove any old entries with the same host
+                same_host_entries = [
+                    entry.entry_id
+                    for entry in self._async_current_entries()
+                    if entry.data[CONF_HOST] == user_input[CONF_HOST]
+                ]
+                if same_host_entries:
+                    await asyncio.wait(
+                        [
+                            self.hass.config_entries.async_remove(entry_id)
```

Comment:
We should not remove existing entries automatically. Show an error, `already_configured`, if the same host is entered.